### PR TITLE
[Tests] Updated instance types used in tests

### PR DESCRIPTION
**Pull request by shawnpanda**
## Why are these changes needed?
- Updated instance types used in tests. GCP instance types `n1-standard-2` and `n2d-standard-2` may be too small and can cause stability issues for ray clusters. Therefore bumping to 4 CPUs
- Changed `i3.8xlarge` (for smoke tests) to `m6i.8xlarge` since that's what the actual release tests use
- Removed yaml files that were never used

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by can-anyscale — APPROVED** [review]

**Comment by can-anyscale** [comment]
I added oncall from all teams: ml, core, data, rllib, serve

File: release/rllib_tests/1gpu_4cpus.yaml [hunk 1]
```diff
@@ -1,17 +0,0 @@
-cloud_id: {{env["ANYSCALE_CLOUD_ID"]}}
-region: us-west-2
-
```
Comment (avnishn) [inline release/rllib_tests/1gpu_4cpus.yaml:3]: this is fine to delete since it wasn't being used.

**Review by avnishn — APPROVED** [review]

**Review by jjyao — APPROVED** [review]

**Comment by shawnpanda** [comment]
Actually `n1-standard-2` and `n2d-standard-2` should still work. I will update the PR so that I don't change them

**Pull request closed without merging**